### PR TITLE
refactor: bring deb_index on par with bzlmod api

### PR DESCRIPTION
**Pull request by thesayyn**
- removes workspace file from root workspace and only test with bzlmod
- renames deb_index to `apt.install` to make bzlmod and nonbzlmod api similar.
- updates CI to test against Bazel 8 and Bazel 7 
- drop support for Bazel 6

### CODE REVIEW

File: MODULE.bazel [hunk 1]
```diff
@@ -6,8 +6,11 @@ module(
     compatibility_level = 1,
 )
 
+# Bazel 8 fails without a working version of protobuf
+bazel_dep(name = "protobuf", version = "29.0-rc2.bcr.1")
 bazel_dep(name = "bazel_skylib", version = "1.5.0")
-bazel_dep(name = "aspect_bazel_lib", version = "2.7.9")
+bazel_dep(name = "aspect_bazel_lib", version = "2.9.4")
+bazel_dep(name = "rules_java", version = "8.3.2")
```
Comment (jbedard) [inline MODULE.bazel:13]: These are all non-dev deps?

File: MODULE.bazel.lock [hunk 2]
```diff
@@ -4,26 +4,41 @@
     "https://bcr.bazel.build/bazel_registry.json": "8a28e4aff06ee60aed2a8c281907fb8bcbf3b753c91fb5a5c57da3215d5b3497",
     "https://bcr.bazel.build/modules/abseil-cpp/20210324.2/MODULE.bazel": "7cd0312e064fde87c8d1cd79ba06c876bd23630c83466e9500321be55c96ace2",
     "https://bcr.bazel.build/modules/abseil-cpp/20211102.0/MODULE.bazel": "70390338f7a5106231d20620712f7cccb659cd0e9d073d1991c038eb9fc57589",
-    "https://bcr.bazel.build/modules/abseil-cpp/20211102.0/source.json": "7e3a9adf473e9af076ae485ed649d5641ad50ec5c11718103f34de03170d94ad",
```
Comment (jbedard) [inline MODULE.bazel.lock:7]: I don't think this file should be checked in? Looks like it already was though 🤷‍♂️

File: apt/BUILD.bazel [hunk 3]
```diff
@@ -15,8 +15,8 @@ bzl_library(
 )
 
 bzl_library(
-    name = "index",
-    srcs = ["index.bzl"],
+    name = "apt",
```
Comment (jbedard) [inline apt/BUILD.bazel:18]: We're ok with beginning changes?

File: MODULE.bazel [hunk 4]
```diff
@@ -20,9 +23,66 @@ use_repo(bazel_lib_toolchains, "yq_linux_ppc64le")
...
 bazel_dep(name = "bazel_skylib_gazelle_plugin", version = "1.5.0", dev_dependency = True)
 bazel_dep(name = "buildifier_prebuilt", version = "6.1.2", dev_dependency = True)
 bazel_dep(name = "platforms", version = "0.0.10", dev_dependency = True)
 bazel_dep(name = "rules_oci", version = "2.0.0-rc0", dev_dependency = True)
 bazel_dep(name = "container_structure_test", version = "1.16.0", dev_dependency = True)
+bazel_dep(name = "bazel_features", version = "1.20.0", dev_dependency = True)
+
+http_archive = use_repo_rule("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+http_archive(
```
Comment (jbedard) [inline MODULE.bazel:37]: Should these be marked as dev?

**Review by jbedard — APPROVED** [review]

File: MODULE.bazel.lock (hunk 2 above)
Comment (thesayyn) [reply MODULE.bazel.lock:7]: Yeah, it's fine to check this in imho.

File: MODULE.bazel (hunk 4 above)
Comment (thesayyn) [reply MODULE.bazel:37]: for some reason use_repo_rule rejects dev_dependency attribute.

File: MODULE.bazel (hunk 1 above)
Comment (thesayyn) [reply MODULE.bazel:13]: Yes

**Pull request merged**